nova/tests/unit/scheduler/filters/test_bigvm_filters.py

```diff
@@ -231,141 +232,110 @@ def call(a, b):
-                          {'full': '1', 'half': '0.5'}, 'filter_scheduler')
         spec_obj = objects.RequestSpec(
             flavor=objects.Flavor(memory_mb=CONF.bigvm_mb,
-                                  extra_specs={'host_fraction': 'broken,half'},
+                                  extra_specs={'host_fraction': 'broken,0.5'},
                                   name='random-name'))
         host = fakes.FakeHostState('host1', 'compute',
-                {'uuid': uuidsentinel.host1})
+                                   {'uuid': uuidsentinel.host1})
         self.filt_cls._HV_SIZE_CACHE[host.uuid] = CONF.bigvm_mb * 2 + 1024
         self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
 
     def test_extra_specs_half_negative(self):
         """test specified half size"""
-        CONF.set_override('bigvm_host_size_filter_uses_flavor_extra_specs',
-                          True, 'filter_scheduler')
-        CONF.set_override('bigvm_host_size_filter_host_fractions',
-                          {'full': '1', 'half': '0.5'}, 'filter_scheduler')
         spec_obj = objects.RequestSpec(
             flavor=objects.Flavor(memory_mb=CONF.bigvm_mb,
-                                  extra_specs={'host_fraction': 'half'},
+                                  extra_specs={'host_fraction': '0.5'},
                                   name='random-name'))
         host = fakes.FakeHostState('host1', 'compute',
-                {'uuid': uuidsentinel.host1})
+                                   {'uuid': uuidsentinel.host1})
         self.assertFalse(self.filt_cls.host_passes(host, spec_obj))
+
+    def test_extra_specs_various_positive(self):
+        """test various arbitrary sizes"""
+        spec_obj = objects.RequestSpec(
+            flavor=objects.Flavor(memory_mb=CONF.bigvm_mb,
+                                  extra_specs={'host_fraction': '0.1'}))
+        host = fakes.FakeHostState('host1', 'compute',
+                                   {'uuid': uuidsentinel.host1})
+
+        self.filt_cls._HV_SIZE_CACHE[host.uuid] = CONF.bigvm_mb * 10 + 1024
+        self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
+
+        spec_obj.flavor.extra_specs['host_fraction'] = '0.2'
+        self.filt_cls._HV_SIZE_CACHE[host.uuid] = CONF.bigvm_mb * 5 + 1024
+        self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
+
+        spec_obj.flavor.extra_specs['host_fraction'] = '0.4'
+        self.filt_cls._HV_SIZE_CACHE[host.uuid] = CONF.bigvm_mb * 2.5 + 1024
+        self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
+
+        spec_obj.flavor.extra_specs['host_fraction'] = '0.6'
+        self.filt_cls._HV_SIZE_CACHE[host.uuid] = CONF.bigvm_mb * 1.67 + 1024
+        self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
+
+        spec_obj.flavor.extra_specs['host_fraction'] = '0.8'
+        self.filt_cls._HV_SIZE_CACHE[host.uuid] = CONF.bigvm_mb * 1.25 + 1024
+        self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
```

Comment:
I don't really see a value in those test-cases. They don't seem to test any edge-case really. Can you elaborate on your intention here? I would understand, if you'd test that really all values from the `host_fraction` setting can be used, by adding all those fractions into the flavor at once and then test it against different host sizes, but can't see the point in doing it like this.

Reply:
I considered it valuable to have test-cases with other-than-0.5-or-1 values for ratios, in case a specific mostly-used list emerges through use in production, which might then be reflected in future code by never checking different values than the ones customarily used.

I agree they're not strictly necessary and these values are a bit arbitrary themselves. There's evidently no method for checking _all_ values, but a `range(0, 1, 0.2)` seemed like a compromise. Can remove it... But personally would leave it in. I can change it to add all values at once.

Reply:
If I understand you correctly, these tests are supposed to guard against us basically hardcoding values in the algorithm. Then imho we should use a random number here and compute the fitting host-fraction in the test.

Reply:
I thought about random numbers, but thought they were a bit of a smell in tests, because then one test run might fail and the next succeed. We could pre-seed the RNG, though.

Reply:
You can compute the expected host-memory necessary for that random number and set it. What this would check, is that any random number passes the algorithm which means there's an algorithm and no hardcoded map or something. I don't really think that's test-worthy, but who knows.

Reply:
Okay, after some discussion I will remove the test:
1. It tests for programmer maliciousness more than anything. It checks against hardcoded values in the algorithgm, which are unlikely to be introduced by anyone mildly competent.
1. It doesn't need to test the parseability of various numbers. Python-`float()` supposedly works fine.
1. Less code